database/redis/tag.go

```diff
@@ -44,6 +44,41 @@ func (connector *DbConnector) GetTagTriggerIDs(tagName string) ([]string, error)
 	return triggerIDs, nil
 }
 
+// CleanUpAbandonedTags deletes tags for which triggers and subscriptions don't exist.
+// Returns count of deleted tags.
+func (connector *DbConnector) CleanUpAbandonedTags() (int, error) {
+	var count int
+	client := *connector.client
+
+	iter := client.SScan(connector.context, tagsKey, 0, "*", 0).Iterator()
+	for iter.Next(connector.context) {
+		tag := iter.Val()
+
+		result, err := client.Exists(connector.context, tagTriggersKey(tag)).Result()
+		if err != nil {
+			return 0, fmt.Errorf("failed to check tag triggers existence, error: %v", err)
+		}
+		isTriggerExists := result == 1
```

Comment:
```suggestion
		if result == 1 { 
		     continue;
		}
```
We don't have to check existance of subscription if the trigger exists, right?

Reply:
It is fine. I don't follow premature optimization. That code executes pretty fast on our data

Reply:
I think, we should use continue here primarily to improve redability

Reply:
Maybe it is a subjective thing. If you think that it would make code more complicated -- let's keep it the way it is

Reply:
🤔  But do we need to improve readability? I think this code is simple enough

Reply:
In my opinion, it is almost always worth it to improve readablility

Reply:
To be precise I think you mean `continue` here is not for readability, but for better understanding. Current code is understandable enough. I try to not to be perfectionist. Also I think cli service in Moira is less important than other, it is kind a helper, so we don't need pay to much attention to it

Reply:
Sorry, we unfortunately spend more time on this debate, than we would on applying these changes to the code. I don't really understand why do you resist so much :(

Reply:
> Sorry, we unfortunately spend more time on this debate, than we would on applying these changes to the code.

I agree with you. 

>  I don't really understand why do you resist so much :(

Because I don't see any profit of suggested change

Reply:
@Tetrergeru I fixed it, check it please https://github.com/moira-alert/moira/pull/773/commits/4554380c75d807618ed4eb9be65a42aaa2899a4f